freqtrade/freqtradebot.py

```diff
@@ -1375,3 +1392,21 @@ def fee_detection_from_trades(self, trade: Trade, order: Dict, order_amount: flo
                                               amount=amount, fee_abs=fee_abs)
         else:
             return amount
+
+    def get_valid_price(self, custom_price: float, proposed_price: float) -> float:
+        """
+        Return the valid price.
+        Check if the custom price is of the good type if not return proposed_price
+        :return: valid price for the order
+        """
+        if custom_price:
+            if isinstance(custom_price, int):
+                valid_price = float(custom_price)
+            elif isinstance(custom_price, float):
+                valid_price = custom_price
+            else:
+                valid_price = proposed_price
```

Comment:
i think a try/except block trying to parse to float directly will be more reliable.

assuming someone (by accident) returns a price as string instead of float - that would handle this case too.

Reply:
Yes I will change that to a try catch method.
Concerning the fact that the user can define a price really far away from the current price, I don't think it's an obligation cap the price.

In fact I plan to set a price really lower If for example if the current price is not where I wanted it to be when i decided to buy.
That way I can perform a sort of "soft abort" of the entry.

Reply:
Maybe whe should log a warning like " Your entry/exit price is XX% lower of the current price, your order have a low probability of getting filled"

Reply:
it's honestly just a thought (basically asking for your opinion) - not a requirement.
i'd not use that as "soft abort" - but use the proper method for that ...

Reply:
Yes I know it's better to have a function for that like "last_moment_abort". But for the moment setting an entry price far away is a temporary solution ;-)

Reply:
You really should read the advanced strategy documentation :laughing: 

We do have `confirm_trade_entry()` - which is made specifically for this.

Reply:
Hahaha my bad, I always thought this method was about a user confirmation with telegram

Reply:
i've just played with this - and "intentionally" placed the orders 10% below current price.

This makes it seem like I've a trade open with immediate 10% profits (it'll report that way until the order is filled) - but it'll then cancel the order as it's really been a bullshit order.

As security measure, I'd propose to add a "allowed range" that defaults to 1-2% (we can make it a parameter so if you explicitly want this, you can go beyond that value). 
Especially your example (using an indicator for pricing) can otherwise result in prices that are far off spot and resut in almost all orders timing out.

Reply:
I'm agree with the security 👍🏽

Reply:
Maybe we should display profit only if order is filled to limit confusion

Reply:
Yeah, but thats a different topic, as youbcan also configure orderbook to almost never fill ...

Reply:
> As security measure, I'd propose to add a "allowed range" that defaults to 1-2% (we can make it a parameter so if you explicitly want this, you can go beyond that value).
> Especially your example (using an indicator for pricing) can otherwise result in prices that are far off spot and resut in almost all orders timing out.

If the user set a price outside the allowed range I think we should cancel the order. It seem dangerous to change the price for him.

Reply:
if you work on that premise, then you'll have to also cancel it in case of None or non-float values.

To me, a min/max limit around propose_price (which can be changed via config setting) will be fine (it'll also be in the documentation, so i don't see a problem with that).

Reply:
> if you work on that premise, then you'll have to also cancel it in case of None or non-float values. You are right

I understand your point but it seem weird to me.
It's like if I go buy a car priced at 10 000$ I make an offer at 9000$, the bailiff tell me that the minimum bid is 9500$ and change my offer to 9500. 
Maybe that car is a good deal at 9000$ buy not at 9500$

Reply:
custom-order-pricing logic will not cancel order-placement.

We have `confirm_trade_entry` for that (which will get the final price as well - so the user can decide there to not buy instead).

If you think "any" price is fine (also for non-experienced users) - then we can leave this "bracket" safety net out.

Reply:
> We have `confirm_trade_entry` for that (which will get the final price as well - so the user can decide there to not buy there). Yes that's right

Reply:
@xmatthias  do you know why there is failing tests on mac os? They does not seem related my modifications

Reply:
After a pull from develop the issue with rpc_telegram tests of mac os has disappeared